### ISSUE TITLE
Отладить пайплайн классификатора unsloth

### ISSUE THREAD

**Issue by pavel-collab** [pavel-collab/SMM4H-2025#7]
Провести полноценное обучение с помощью скрипта + скрипт инференса (там в конце надописанная часть скрипта -- нужно дописать)

**Comment by pavel-collab** [comment]
Есть небольшая проблемка. Скрипты работают только при определенных версиях библиотек. В противном случае пайплайн отрабатывает плохо. Модель не дает ожиаемого ответа. 

Дамп набора пакетов

absl-py==2.2.2
accelerate==1.8.1
aiohappyeyeballs==2.6.1
aiohttp==3.11.18
aiosignal==1.3.2
airportsdata==20250224
annotated-types==0.7.0
anyio==4.9.0
astor==0.8.1
asttokens==3.0.0
attrs==25.3.0
beautifulsoup4==4.13.4
bitsandbytes==0.46.0
blake3==1.0.4
bs4==0.0.2
cachetools==5.5.2
certifi==2025.4.26
charset-normalizer==3.4.2
click==8.2.0
cloudpickle==3.1.1
comm==0.2.2
compressed-tensors==0.9.3
contourpy==1.3.2
cupy-cuda12x==13.4.1
cut-cross-entropy==25.1.1
cycler==0.12.1
datasets==3.6.0
debugpy==1.8.14
decorator==5.2.1
Deprecated==1.2.18
depyf==0.18.0
diffusers==0.34.0
dill==0.3.8
diskcache==5.6.3
distro==1.9.0
dnspython==2.7.0
docstring_parser==0.16
einops==0.8.1
email_validator==2.2.0
emoji==1.4.1
executing==2.2.0
fastapi==0.115.12
fastapi-cli==0.0.7
fastrlock==0.8.3
filelock==3.18.0
fonttools==4.58.0
frozenlist==1.6.0
fsspec==2025.3.0
gguf==0.16.3
googleapis-common-protos==1.70.0
grpcio==1.71.0
h11==0.16.0
hf-xet==1.1.0
hf_transfer==0.1.9
httpcore==1.0.9
httptools==0.6.4
httpx==0.28.1
huggingface-hub==0.31.1
idna==3.10
importlib_metadata==8.0.0
interegular==0.3.3
ipykernel==6.29.5
ipython==9.2.0
ipython_pygments_lexers==1.1.1
jedi==0.19.2
Jinja2==3.1.6
jiter==0.9.0
joblib==1.5.0
jsonschema==4.23.0
jsonschema-specifications==2025.4.1
jupyter_client==8.6.3
jupyter_core==5.7.2
kiwisolver==1.4.8
lark==1.2.2
llama_cpp_python==0.3.13
llguidance==0.7.19
llvmlite==0.44.0
lm-format-enforcer==0.10.11
Markdown==3.8
markdown-it-py==3.0.0
MarkupSafe==3.0.2
matplotlib==3.10.3
matplotlib-inline==0.1.7
mdurl==0.1.2
mistral_common==1.5.4
mpmath==1.3.0
msgpack==1.1.0
msgspec==0.19.0
multidict==6.4.3
multiprocess==0.70.16
nest-asyncio==1.6.0
networkx==3.4.2
ninja==1.11.1.4
numba==0.61.2
numpy==2.2.5
nvidia-cublas-cu12==12.4.5.8
nvidia-cuda-cupti-cu12==12.4.127
nvidia-cuda-nvrtc-cu12==12.4.127
nvidia-cuda-runtime-cu12==12.4.127
nvidia-cudnn-cu12==9.1.0.70
nvidia-cufft-cu12==11.2.1.3
nvidia-cufile-cu12==1.11.1.6
nvidia-curand-cu12==10.3.5.147
nvidia-cusolver-cu12==11.6.1.9
nvidia-cusparse-cu12==12.3.1.170
nvidia-cusparselt-cu12==0.6.2
nvidia-nccl-cu12==2.21.5
nvidia-nvjitlink-cu12==12.4.127
nvidia-nvtx-cu12==12.4.127
openai==1.78.1
opencv-python-headless==4.11.0.86
opentelemetry-api==1.26.0
opentelemetry-exporter-otlp==1.26.0
opentelemetry-exporter-otlp-proto-common==1.26.0
opentelemetry-exporter-otlp-proto-grpc==1.26.0
opentelemetry-exporter-otlp-proto-http==1.26.0
opentelemetry-proto==1.26.0
opentelemetry-sdk==1.26.0
opentelemetry-semantic-conventions==0.47b0
opentelemetry-semantic-conventions-ai==0.4.8
outlines==0.1.11
outlines_core==0.1.26
packaging==25.0
pandas==2.2.3
parso==0.8.4
partial-json-parser==0.2.1.1.post5
peft==0.15.2
pexpect==4.9.0
pillow==11.2.1
platformdirs==4.3.8
prometheus-fastapi-instrumentator==7.1.0
prometheus_client==0.21.1
prompt_toolkit==3.0.51
propcache==0.3.1
protobuf==3.20.3
psutil==7.0.0
ptyprocess==0.7.0
pure_eval==0.2.3
py-cpuinfo==9.0.0
pyarrow==20.0.0
pycountry==24.6.1
pydantic==2.11.4
pydantic_core==2.33.2
Pygments==2.19.1
pyparsing==3.2.3
python-dateutil==2.9.0.post0
python-dotenv==1.1.0
python-json-logger==3.3.0
python-multipart==0.0.20
pytz==2025.2
PyYAML==6.0.2
pyzmq==26.4.0
ray==2.46.0
referencing==0.36.2
regex==2024.11.6
requests==2.32.3
rich==14.0.0
rich-toolkit==0.14.6
rpds-py==0.24.0
safetensors==0.5.3
scikit-learn==1.6.1
scipy==1.15.3
seaborn==0.13.2
sentencepiece==0.2.0
setuptools==80.4.0
shellingham==1.5.4
shtab==1.7.2
six==1.17.0
sniffio==1.3.1
soupsieve==2.7
stack-data==0.6.3
starlette==0.46.2
sympy==1.13.1
tensorboard==2.19.0
tensorboard-data-server==0.7.2
threadpoolctl==3.6.0
tiktoken==0.9.0
tokenizers==0.21.2
torch==2.6.0
torchaudio==2.6.0
torchvision==0.21.0
tornado==6.4.2
tqdm==4.67.1
traitlets==5.14.3
transformers==4.53.3
triton==3.2.0
trl==0.19.0
typeguard==4.4.4
typer==0.15.3
typing-inspection==0.4.0
typing_extensions==4.14.0
tyro==0.9.24
tzdata==2025.2
unsloth @ git+https://github.com/unslothai/unsloth.git@0eb61fbea728cdc8acd1f2fa1f6f71074f559ac0
unsloth_zoo @ git+https://github.com/unslothai/unsloth-zoo.git@cfaaaa377d8da6115bd7c8ce66f458821225ff7f
urllib3==2.4.0
uvicorn==0.34.2
uvloop==0.21.0
vllm==0.8.5.post1
watchfiles==1.0.5
wcwidth==0.2.13
websockets==15.0.1
Werkzeug==3.1.3
wheel==0.45.1
wrapt==1.17.2
xformers==0.0.29.post3
xgrammar==0.1.18
xxhash==3.5.0
yarl==1.20.0
zipp==3.21.0
